joern-cli/frontends/x2cpg/src/main/scala/io/joern/x2cpg/SourceFiles.scala

```diff
@@ -70,17 +74,60 @@ object SourceFiles {
     def hasSourceFileExtension(file: File): Boolean =
       file.extension.exists(sourceFileExtensions.contains)
 
-    val (dirs, files) = inputPaths
-      .map(File(_))
-      .partition(_.isDirectory)
+    val inputFiles = inputPaths.map(File(_))
+    assertAllExist(inputFiles)
+
+    val (dirs, files) = inputFiles.partition(_.isDirectory)
 
     val matchingFiles = files.filter(hasSourceFileExtension).map(_.toString)
     val matchingFilesFromDirs = dirs
       .flatMap(_.listRecursively(VisitOptions.follow))
       .filter(hasSourceFileExtension)
       .map(_.pathAsString)
 
-    (matchingFiles ++ matchingFilesFromDirs).toList.sorted
+    val sourceFiles = (matchingFiles ++ matchingFilesFromDirs).toList.sorted
+    assertNonEmpty(inputPaths, sourceFiles)
+    sourceFiles
+  }
+
+  /** Attempting to analyse source paths that do not exist is a hard error. Terminate execution early to avoid
+    * unexpected and hard-to-debug issues in the results.
+    */
+  private def assertAllExist(files: Set[File]): Unit = {
+    val (existant, nonExistant) = files.partition(_.isReadable)
+    val nonReadable             = existant.filterNot(_.isReadable)
+
+    if (nonExistant.nonEmpty || nonReadable.nonEmpty) {
+      logErrorWithPaths("Source input paths do not exist", nonExistant.map(_.canonicalPath))
+
+      logErrorWithPaths("Source input paths exist, but are not readable", nonReadable.map(_.canonicalPath))
+
+      throw InvalidInputPathsException
```

Comment:
Please reuse existing Java exceptions if possible. `NoSuchFileException` should be fine here.